reporting/test/webrequest-pipeline/index.test.js

```diff
@@ -58,5 +60,88 @@ describe('WebRequestPipeline', function () {
         });
       });
     });
+
+    context('with pre-recorded events', function () {
+      let pipeline;
+
+      beforeEach(function () {
+        pipeline = new WebRequestPipeline();
+        pipeline.init();
+        [
+          'onBeforeRequest',
+          'onBeforeSendHeaders',
+          'onHeadersReceived',
+          'onAuthRequired',
+          'onBeforeRedirect',
+          'onCompleted',
+          'onErrorOccurred',
+        ].forEach((step) => {
+          pipeline.addPipelineStep(step, {
+            name: 'test',
+            spec: 'annotate',
+            fn: () => {},
+          });
+        });
+      });
+
+      afterEach(function () {
+        pipeline.unload();
+        pipeline = undefined;
+      });
+
+      context('0001-quick-close', function () {
+        it('runs without a crash', async function () {
+          await playScenario(chrome, {
```

Comment:
One drawback of separating steps, recorded browser event logs, and expectations is that it requires back-and-forth jumping to get an intuition what the test is doing. Maybe putting links to the description as comment would help to make that smoother?

Having a comment with a URL like that: 
https://github.com/ghostery/webextension-event-recorder/blob/main/scenariors/0001-quick-close.js